### PR TITLE
Multiregion

### PR DESCRIPTION
**Pull request by obrestad**
This release bring in all changes needed to support the multi-region architecture of our openstack platforms.

### Changes

- The openstack-plugin from zabbix that scrapes the openstack-databases for operational-data is now updated to support getting the misc project-ID from zabbix instead of reading the keystone database. This is needed in multi-region as the keystone-database is not on the same database-server as for instance nova.
- All keystone-credentials and service-overview is now placed in a common global hiera-key to have one place where regions and services is defined. Here usernames can also be specified to allow using different keystone users for different regions.
- Now databases and haproxy-frontends are only created for the services actually being installed in a certain region.

### New hiera-keys

- `ntnuopenstack::endpoint::admin::cert` - A certificate+key used by the administrative openstack service endpoints.
- `ntnuopenstack::endpoint::public::cert` - A certificate+key used by the public openstack service endpoints.
- `ntnuopenstack::cinder::ceph::user` - The ceph-user cinder should use to manage rbd volumes
- `ntnuopenstack::glance::ceph::pool` - Ceph-pool used by glance
- `ntnuopenstack::glance::ceph::user` - Ceph user used by glance
- `ntnuopenstack::nova::ceph::user` - Ceph user used by nova
- `ntnuopenstack::octavia::management::networks` - CIDRs representing networks where octavia amphorae interfaces is placed.
- `ntnuopenstack::services`: A large hash containing regions with their associated services.
- `profile::networks::openstack::users` - A list of CIDRs representing networks where we allow access to openstack web/APIs from.
- `ntnuopenstack::keystone::region` - This key indicates which regions urls are used to get to keystone.

### Deprecated hiera-keys

- `ntnuopenstack::barbican::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::cinder::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::glance::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::heat::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::magnum::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::neutron::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::nova::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::octavia::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::placement::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::swift::keystone::password` - Replaced by `ntnuopenstack::services`
- `ntnuopenstack::swift::ceph::key` - Replaced by `profile::ceph::keys`

### CODE REVIEW

File: manifests/horizon/plugins.pp [hunk 1]
```diff
@@ -1,36 +1,25 @@
...
-    })
+  $services.each | $region, $data | {
+    if('heat' in $data['services']) {
+      include ::horizon::dashboards::heat
+    }
+    if('octavia' in $data['services']) {
+      include ::horizon::dashboards::octavia
+    }
+    if('magnum' in $data['services']) {
+      if($magnum and $::osfamily == 'RedHat') {
```
Comment (LarsErikP) [inline manifests/horizon/plugins.pp:15]: Kanskje vi kan fjerne RedHat-greiene?

Comment (LarsErikP) [reply manifests/horizon/plugins.pp:15]: Og uansett.. Den her if($magnum...) konstruksjonen blir ikke rett nå? Den variablen er ikke definert lenger

File: manifests/nova/quota.pp [hunk 2]
```diff
@@ -24,7 +25,8 @@
     class { '::nova::limit':
```
Comment (LarsErikP) [inline manifests/nova/quota.pp:25]: Trengs det ikke brukernavn her?

**Review by LarsErikP — COMMENTED** [review]
Med mine få kommentarer, er det sikkert ok =)

File: manifests/horizon/plugins.pp (hunk 1 above)
Comment (obrestad) [reply manifests/horizon/plugins.pp]: Da fjernet vi redhat-greier.

File: manifests/nova/quota.pp (hunk 2 above)
Comment (obrestad) [reply manifests/nova/quota.pp:25]: Jo! Det må vi ha.

File: manifests/horizon/plugins.pp (hunk 1 above)
Comment (LarsErikP) [reply manifests/horizon/plugins.pp]: Brabra!

**Review by LarsErikP — APPROVED** [review]
Very good!

**Comment by LarsErikP** [comment]
Nøkkelen `ntnuopenstack::keystone::region` mangler i oversikt over nye nøkler

**Pull request merged**